cpp/src/unary/math_ops.cu

```diff
@@ -466,13 +519,42 @@ struct LogicalOpDispatcher {
   }
 };
 
+struct FixedPointOpDispatcher {
+  template <typename T, typename... Args>
+  std::enable_if_t<not cudf::is_fixed_point<T>(), std::unique_ptr<column>> operator()(
+    Args&&... args)
+  {
+    CUDF_FAIL("FixedPointOpDispatcher only for fixed_point");
+  }
+
+  template <typename T>
+  std::enable_if_t<cudf::is_fixed_point<T>(), std::unique_ptr<column>> operator()(
+    column_view const& input,
+    cudf::unary_op op,
+    cudaStream_t stream,
+    rmm::mr::device_memory_resource* mr)
+  {
+    // clang-format off
+    switch (op) {
+      case cudf::unary_op::CEIL:  return unary_op_with<T, fixed_point_ceil> (input, stream, mr);
+      case cudf::unary_op::FLOOR: return unary_op_with<T, fixed_point_floor>(input, stream, mr);
+      case cudf::unary_op::ABS:   return unary_op_with<T, fixed_point_abs>  (input, stream, mr);
+      default: CUDF_FAIL("Unsupported fixed_point unary operation");
+    }
+    // clang-format on
```

Comment:
I don't think it is fair to just turn off clang-format when you have a different opinion about how it should be formatting. If we all expressed our formatting opinions by turning off clang-format, the code would have no standards.

Reply:
I agree and I disagree. I agree with:
>I don't think it is fair to just turn off clang-format when you have a different opinion about how it should be formatting. If we all expressed our formatting opinions by turning off clang-format, the code would have no standards.

However, I don't turn off clang-format whenever I disagree. The majority of the time when I disagree, I leave the code be. However, when it comes to **code alignment** that leads to much more visually parse-able code, I think exceptions should be allowed. Due to alignment, differences stand out like a sore thumb, point in case is @hyperbolic2346 catch of an unnecessary `detail::` namespace: https://github.com/rapidsai/cudf/pull/6777#discussion_r525365272

Other cases where I have turned it off:

**Example 1**
```cpp
// clang-format off
class list_scalar;
class string_scalar;
template <typename T> class numeric_scalar;
template <typename T> class fixed_point_scalar;
template <typename T> class timestamp_scalar;
template <typename T> class duration_scalar;

class string_scalar_device_view;
template <typename T> class numeric_scalar_device_view;
template <typename T> class fixed_point_scalar_device_view;
template <typename T> class timestamp_scalar_device_view;
template <typename T> class duration_scalar_device_view;
// clang-format on
```
vs
```cpp
class list_scalar;
class string_scalar;
template <typename T>
class numeric_scalar;
template <typename T>
class fixed_point_scalar;
template <typename T>
class timestamp_scalar;
template <typename T>
class duration_scalar;

class string_scalar_device_view;
template <typename T>
class numeric_scalar_device_view;
template <typename T>
class fixed_point_scalar_device_view;
template <typename T>
class timestamp_scalar_device_view;
template <typename T>
class duration_scalar_device_view;
```

**Example 2**
```cpp
// clang-format off
template <typename T>
using device_storage_type_t =
  std::conditional_t<std::is_same<numeric::decimal32, T>::value, int32_t,
  std::conditional_t<std::is_same<numeric::decimal64, T>::value, int64_t, T>>;
// clang-format on
```
vs
```cpp
template <typename T>
using device_storage_type_t =
  std::conditional_t<std::is_same<numeric::decimal32, T>::value,
                     int32_t,
                     std::conditional_t<std::is_same<numeric::decimal64, T>::value, int64_t, T>>;
```

**Example 3**
```cpp
    // clang-format off
    switch (method) {
      case cudf::rounding_method::HALF_UP:
        if      (is_fixed_point<T>()) return round_with<T, half_up_fixed_point>(input, decimal_places, stream, mr);
        else if (decimal_places == 0) return round_with<T, half_up_zero       >(input, decimal_places, stream, mr);
        else if (decimal_places >  0) return round_with<T, half_up_positive   >(input, decimal_places, stream, mr);
        else                          return round_with<T, half_up_negative   >(input, decimal_places, stream, mr);
      case cudf::rounding_method::HALF_EVEN:
        if      (is_fixed_point<T>()) return round_with<T, half_even_fixed_point>(input, decimal_places, stream, mr);
        else if (decimal_places == 0) return round_with<T, half_even_zero       >(input, decimal_places, stream, mr);
        else if (decimal_places >  0) return round_with<T, half_even_positive   >(input, decimal_places, stream, mr);
        else                          return round_with<T, half_even_negative   >(input, decimal_places, stream, mr);
      default: CUDF_FAIL("Undefined rounding method");
    }
```
vs
```cpp
    switch (method) {
      case cudf::rounding_method::HALF_UP:
        if (is_fixed_point<T>())
          return round_with<T, half_up_fixed_point>(input, decimal_places, stream, mr);
        else if (decimal_places == 0)
          return round_with<T, half_up_zero>(input, decimal_places, stream, mr);
        else if (decimal_places > 0)
          return round_with<T, half_up_positive>(input, decimal_places, stream, mr);
        else
          return round_with<T, half_up_negative>(input, decimal_places, stream, mr);
      case cudf::rounding_method::HALF_EVEN:
        if (is_fixed_point<T>())
          return round_with<T, half_even_fixed_point>(input, decimal_places, stream, mr);
        else if (decimal_places == 0)
          return round_with<T, half_even_zero>(input, decimal_places, stream, mr);
        else if (decimal_places > 0)
          return round_with<T, half_even_positive>(input, decimal_places, stream, mr);
        else
          return round_with<T, half_even_negative>(input, decimal_places, stream, mr);
      default: CUDF_FAIL("Undefined rounding method");
    }
```

Note that there are 38 places in `libcudf` where we turn off clang-format (most not by me), and in most cases they undeniably lead to better aligned more readable code.

Reply:
In this particular case you are saving only 3 lines, and the result would not be that bad looking, IMO. I don't think it is as useful to turn it off here as the similar but different if/else if Example 3.

Also, your change violates the "readable in github without horizontal scrolling" pseudo-rule. :)

Reply:
> the result would not be that bad looking, IMO

I agree. For this case it actually hardly changes things (I think previously it looked a lot worse but the final version is actually OK). I will change it back

>In this particular case you are saving only 3 lines

Just to clarify, it is not about saving lines. In fact example 2 increases line count by 1 if you include the off/on comments. It only has to do with **code alignment** - which makes it more visually parseable.

> Also, your change violates the "readable in github without horizontal scrolling" pseudo-rule. :)

I know :frowning_face: however, I would argue that the increase in readability is definitely worth the 17 character violation.